osu.Game/Screens/SelectV2/Carousel.cs

```diff
@@ -650,6 +654,12 @@ private static void expirePanelImmediately(Drawable panel)
         {
             panel.FinishTransforms();
             panel.Expire();
+
+            var carouselPanel = (ICarouselPanel)panel;
+
+            carouselPanel.Item = null;
+            carouselPanel.Selected.Value = false;
+            carouselPanel.KeyboardSelected.Value = false;
```

Comment:
What's the significance of these operations? Should I be worrying that `CarouselYPosition` isn't reset here?

Reply:
I just see it as sane defaults for animation purposes. ie. new panel comes alive, if it gets selected i'd hope it goes from not-selected to selected, rather than starting with selected==true.

> Should I be worrying that CarouselYPosition isn't reset here?

I assume you mean `DrawYPosition`? I don't know what a sane default for that would be. We could set it back to 0 but I think it's kinda up to carousel to manage that (it already transfers when retrieving from pool immediately)

Reply:
I just noticed that you moved this logic from `BeatmapPanel.FreeAfterUse()` which I guess gives this change more of a point. I can agree with it in that light.